docs/python_docs/python/tutorials/getting-started/crash-course/1-ndarray.md

```diff
@@ -1,121 +1,475 @@
-<!--- software distributed under the License is distributed on an -->
-<!--- "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY -->
-<!--- KIND, either express or implied.  See the License for the -->
-<!--- specific language governing permissions and limitations -->
-<!--- under the License. -->
-
 # Step 1: Manipulate data with NP on MXNet
 
-This getting started exercise introduces the `np` package, which is similar to Numpy. For more information, please see [Differences between NP on MXNet and NumPy](/api/python/docs/tutorials/getting-started/deepnumpy/deepnumpy-vs-numpy.html).
+This getting started exercise introduces the MXNet `np` package for ndarrays.
+These ndarrays extend the functionality of the common NumPy ndarrays. Many NumPy
+methods are also available within MXNet; therefore, we will only briefly cover
+some of what is available to view more you can go [here](www.nohello.com).
 
 ## Import packages and create an array
-
-
-To get started, run the following commands to import the `np` package together with the NumPy extensions package `npx`. Together, `np` with `npx` make up the NP on MXNet front end.
+To get started, run the following commands to import the `np` package together
+with the NumPy extensions package `npx`. Together, `np` with `npx` make up the
+NP on MXNet front end.
 
 ```{.python .input  n=1}
 from mxnet import np, npx
 npx.set_np()  # Activate NumPy-like mode.
 ```
 
-In this step, create a 2D array (also called a matrix). The following code example creates a matrix with values from two sets of numbers: 1, 2, 3 and 4, 5, 6. This might also be referred to as a tuple of a tuple of integers.
+In this step, create a 2D array (also called a matrix). The following code
+example creates a matrix with values from two sets of numbers: 1, 2, 3 and 4, 5,
+6. This might also be referred to as a tuple of a tuple of integers.
 
 ```{.python .input  n=2}
 np.array(((1,2,3),(5,6,7)))
 ```
 
-You can also create a very simple matrix with the same shape (2 rows by 3 columns), but fill it with 1s.
+```{.json .output n=2}
+[
+ {
+  "data": {
+   "text/plain": "array([[1., 2., 3.],\n       [5., 6., 7.]])"
+  },
+  "execution_count": 2,
+  "metadata": {},
+  "output_type": "execute_result"
+ }
+]
+```
+
+You can also create a very simple matrix with the same shape (2 rows by 3
+columns), but fill it with 1's.
 
 ```{.python .input  n=3}
-x = np.ones((2,3))
+x = np.full((2,3),1) 
 x
 ```
 
-You can create arrays whose values are sampled randomly. For example, sampling values uniformly between -1 and 1. The following code example creates the same shape, but with random sampling.
+```{.json .output n=3}
+[
+ {
+  "data": {
+   "text/plain": "array([[1., 1., 1.],\n       [1., 1., 1.]])"
+  },
+  "execution_count": 3,
+  "metadata": {},
+  "output_type": "execute_result"
+ }
+]
+```
 
-```{.python .input  n=15}
+Alternatively, you could use a command to generate ones.
```

Comment:
let's make sure we align on the terminology. `np.ones` is an array creation routine